src/Tutor.Core/DomainModel/Feedback/FeedbackService.cs

```diff
@@ -0,0 +1,31 @@
+﻿using FluentResults;
+using System.Collections.Generic;
+using Tutor.Core.DomainModel.KnowledgeComponents;
+
+namespace Tutor.Core.DomainModel.Feedback
+{
+    public class FeedbackService : IFeedbackService
+    {
+        private readonly IFeedbackRepository _feedbackRepository;
+        private readonly IKCRepository _kcRepository;
+
+        public FeedbackService(IFeedbackRepository feedbackRepository, IKCRepository kCRepository)
+        {
+            _feedbackRepository = feedbackRepository;
+            _kcRepository = kCRepository;
+        }
+
+        public Result<EmotionsFeedback> SaveEmotionsFeedback(EmotionsFeedback emotionsFeedback)
+        {
+            if (emotionsFeedback.Comment == null)
+                return Result.Fail("Empty Emotions Feedback");
+         
+            var knowledgeComponent = _kcRepository.GetKnowledgeComponent(emotionsFeedback.KnowledgeComponentId);
```

Comment:
You don't need this retrieval and check for the KC. You are not performing it for the learner, and in general it is unnecessary.